pkg/cloud/services/iam/reconcilers.go

```diff
@@ -0,0 +1,180 @@
+package iam
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	v1certmanager "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/cmd/clusterawsadm/converters"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// reconcilePodIdentityWebhook generates certs and starts the webhook in the workload cluster
+// https://github.com/aws/amazon-eks-pod-identity-webhook
+// 1. generate webhook certs via cert-manager in the management cluster
+// 2. push cert secret down to the workload cluster
+// 3. deploy pod identity webhook components with mounted certs (rbac,deployment,mwh,service).
+func (s *Service) reconcilePodIdentityWebhook(ctx context.Context) error {
+	certName := fmt.Sprintf(PodIdentityWebhookCertificateFormat, s.scope.Name())
+	certSecret, err := certificateSecret(ctx,
+		certName, s.scope.Namespace(),
+		fmt.Sprintf(SelfsignedIssuerFormat, s.scope.Name()), []string{
+			fmt.Sprintf("%s.%s.svc", podIdentityWebhookName, s.scope.Namespace()),
+			fmt.Sprintf("%s.%s.svc.cluster.local", podIdentityWebhookName, s.scope.Namespace()),
+		}, s.scope.ManagementClient())
+
+	if err != nil {
+		return err
+	}
+
+	remoteClient, err := s.scope.RemoteClient()
+	if err != nil {
+		return err
+	}
+
+	// switch it to kube-system and move it to the remote cluster
+	if err := reconcileCertificateSecret(ctx, certSecret, remoteClient); err != nil {
+		return err
+	}
+
+	if err := reconcilePodIdentityWebhookComponents(ctx, s.scope.Namespace(), certSecret, remoteClient); err != nil {
```

Comment:
The use of `s.scope.Namespace()` here means that CAPA will attempt to install the pod-identity-webhook into whatever namespace the cluster resources where created in, on the management cluster.

This works for the simplest case of installing everything into `default`, but if the cluster resources are created in something else like a namespace called `capi-<cluster-name>` then this will fail, because the namespace does not exist in the remote cluster. 

IMO - the pod-identity-webhook should be deployed into the `kube-system` namespace.

Reply:
I've pushed a commit [here](https://github.com/sl1pm4t/cluster-api-provider-aws/commit/a2b8fb94338a8fc600ae233b02da0a374de1e09f) that:
* hard codes the installation namespace of pod-identity-webhook to be `kube-system`
* fixes the certificates to reflect the new namespace
* fixes an issue that could prevent a successful reconcile if the webhook certificate secret already existed in the remote cluster (from a previous reconcile loop).
* bumps the `pod-identity-webhook` image version to the latest (v0.5.5)

Reply:
this was based on the install examples of the podid wh... https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/deploy/deployment-base.yaml#L5 trying to keep the install as similar to upstream as possible which does install it into default. the rig I used to work on made that namespace for you in the workload cluster (we created and pivoted and put everything there) so I never caught that bug,

I'm ok with hardcoding to `kube-system` (which is probably the best place for it) but just like to make a case for `default` being where the AWS/EKS devs intended self-install users to put it

Reply:
I've always thought it was strange their examples deployed to `default`. From an permissions point of view, it's easy to protect access to it in `kube-system` but often users would have permission in `default`. 

I note that kOps also hard codes it to `kube-system`: https://github.com/kubernetes/kops/blob/41480d95a96edad247ecd777b2790ecda33f6a85/upup/models/cloudup/resources/addons/eks-pod-identity-webhook.addons.k8s.io/k8s-1.16.yaml.template#L69